Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp

```diff
@@ -1943,7 +1943,8 @@ void MediaPlayerPrivateGStreamer::handleMessage(GstMessage* message)
             // Let the mediaPlayerClient handle the stream error, in this case the HTMLMediaElement will emit a stalled event.
             GST_ERROR("Decode error, let the Media element emit a stalled event.");
             m_loadingStalled = true;
-            break;
+            error = MediaPlayer::NetworkState::DecodeError;
```

Comment:
If you're triggering a DecodeError here, the code block is equivalent to the one in the `else if` block. Please, merge both blocks into a single one, triggered by an OR (`||`) condition.